src/settings.cpp

```diff
@@ -2067,6 +2080,10 @@ const SettingDesc *GetSettingFromName(const char *name)
 		if (!SlIsObjectCurrentlyValid(sd->save.version_from, sd->save.version_to)) continue;
 		if (strcmp(sd->name, name) == 0) return sd.get();
 	}
+	for (auto &sd : _secrets_settings) {
```

Comment:
Please put the existing code, except the bit for the company below, in a separate function with `const SettingTable &` as parameter and iterate over the settings in the parameter. Now you're duplicating 20 lines of code just to do exactly the same but with another collection.
Note: I think it is fine to run the full name and short name check on `_settings` first, and then do the same on `_secret_settings` so you only need one function and two calls to it. After all, any setting with a short name that exists in both the _settings as _secret_settings seems like a bug; why is there something with a name that implies it requires secrecy in the non-secret settings?